pkg/repository/v1manifest/local_manifests.go

```diff
@@ -51,14 +59,39 @@ type LocalManifests interface {
 	Keys() crypto.KeyStore
 }
 
+func (ms *FsManifests) initKeys() error {
+	var root Root
+	_, err := ms.LoadManifest(&root)
+	if err != nil {
+		return errors.AddStack(err)
+	}
+
+	keys, err := root.GetRootKeyStore()
+	if err != nil {
+		return errors.AddStack(err)
+	}
+
+	ms.keys = keys
+	return nil
+}
+
 // SaveManifest implements LocalManifests.
 func (ms *FsManifests) SaveManifest(manifest *Manifest, filename string) error {
 	bytes, err := json.Marshal(manifest)
 	if err != nil {
 		return err
 	}
 
-	return ioutil.WriteFile(filepath.Join(ms.root, filename), bytes, 0644)
+	err = ioutil.WriteFile(filepath.Join(ms.root, filename), bytes, 0644)
+	if err != nil {
+		return err
+	}
+
+	if manifest.Signed.Base().Ty == ManifestTypeRoot {
+		return ms.initKeys()
+	}
```

Comment:
I think this should be in v1repository rather than here - the behaviour should be the same independent of where the manifests are coming from